packages/flutter/lib/src/animation/animations.dart

```diff
@@ -533,3 +533,66 @@ class TrainHoppingAnimation extends Animation<double>
     return '$currentTrain\u27A9$runtimeType(no next)';
   }
 }
+
+/// An interface for combining multiple Animations. Subclasses need only
+/// implement the `value` getter to control how the child animations are
+/// combined. Can be chained to combine more than 2 animations.
+/// Example: to create an animation that is the sum of two others, subclass
+/// this class and define value = first.value + second.value;
+abstract class CompoundAnimation<T> extends Animation<T>
+  with AnimationLazyListenerMixin, AnimationLocalListenersMixin, AnimationLocalStatusListenersMixin {
+  /// Creates a CompoundAnimation. Both arguments must be non-null. Either can
+  /// be a CompoundAnimation itself to combine multiple animations.
+  CompoundAnimation({
+    this.first,
+    this.next,
+  }) {
+    assert(first != null);
+    assert(next != null);
+  }
+
+  /// The first sub-animation. Its status takes precedence if neither are
+  /// animating.
+  final Animation<T> first;
+
+  /// The second sub-animation.
+  final Animation<T> next;
+
+  @override
+  void didStartListening() {
+    first.addListener(notifyListeners);
+    first.addStatusListener(_maybeNotifyStatusListeners);
+    next.addListener(notifyListeners);
+    next.addStatusListener(_maybeNotifyStatusListeners);
+  }
+
+  @override
+  void didStopListening() {
+    first.removeListener(notifyListeners);
+    first.removeStatusListener(_maybeNotifyStatusListeners);
+    next.removeListener(notifyListeners);
+    next.removeStatusListener(_maybeNotifyStatusListeners);
+  }
+
+  @override
+  AnimationStatus get status {
+    // If one of the sub-animations is moving, use that status. Otherwise,
+    // default to `first`.
+    if (next.status == AnimationStatus.forward || next.status == AnimationStatus.reverse)
+      return next.status;
+    return first.status;
+  }
+
+  @override
+  String toString() {
+    return '$runtimeType($first, $next)';
+  }
+
+  AnimationStatus _lastStatus;
+  void _maybeNotifyStatusListeners(AnimationStatus _) {
```

Comment:
It's a little confusing because this argument is normally called status, and there is an identifier with that name in scope here that you do use, it's just not the argument. I dunno what to suggest. I worry that one day I'll be going through the code and will change this to "status" reflexively so that the code is more self-documenting, and will thus break the method (because now it'll mean something else).

Maybe what that tells us is that what we really need here is a test of this class, in particular a test that verifies that this function works as expected. :-)